src/Microsoft.Data.Analysis.Interactive/DataFrameKernelExtension.cs

```diff
@@ -1,47 +1,140 @@
+
+        public string GetFriendlyName(Type type)
         {
-            //Formatter<DataFrame>.Register((tree, writer) =>
-            //{
-            //    writer.Write("");
-            //}, "text/html");
+            string friendlyName = type.Name;
+            if (type.IsArray)
+            {
+                // Not handled yet
+                return "DataFrameColumn";
+            }
+            if (type.IsGenericType)
+            {
+                int backTick = friendlyName.IndexOf('`');
+                if (backTick > 0)
+                {
+                    friendlyName = friendlyName.Remove(backTick);
+                }
+                friendlyName += "<";
+                Type[] typeParameters = type.GetGenericArguments();
+                for (int i = 0; i < typeParameters.Length; ++i)
+                {
+                    string typeParamName = GetFriendlyName(typeParameters[i]);
+                    friendlyName += (i == 0 ? typeParamName : "," + typeParamName);
+                }
+                friendlyName += ">";
+            }
+
+            return friendlyName;
+        }
+
+        public string GetTypedDataFrameWithProperties(DataFrame dataFrame, string resultTypeName, out StringBuilder prettyFormatter)
+        {
+            prettyFormatter = new StringBuilder();
+            StringBuilder stringBuilder = new StringBuilder();
+            string constructor = @$"
+public class {resultTypeName} : DataFrame 
+{{
+    public {resultTypeName}(DataFrame dataFrame)
+    {{
+        foreach (var column in dataFrame.Columns)
+        {{
+            Columns.Add(column);
+        }}
+    }}
+
+";
+            stringBuilder.Append(constructor);
+            prettyFormatter.Append(constructor);
 
+            foreach (var column in dataFrame.Columns)
+            {
+                string columnName = column.Name.Replace(" ", string.Empty);
+                Type dataType = column.DataType;
+                string typeName = GetFriendlyName(column.GetType());
+                stringBuilder.Append($@"
+    public {typeName} {columnName} 
+    {{
+        get
+        {{
+            int columnIndex = Columns.IndexOf(""{columnName}"");
+            return Columns[columnIndex] as {typeName};
+        }}
+    }}");
+                stringBuilder.AppendLine();
+                prettyFormatter.Append($@"
+    public {typeName} {columnName} {{ get; }}");
+                prettyFormatter.AppendLine();
+            }
+
+            stringBuilder.AppendLine();
+            stringBuilder.Append(@"}");
+            prettyFormatter.AppendLine();
+            prettyFormatter.Append(@"}");
+            prettyFormatter.AppendLine();
+            return stringBuilder.ToString();
+        }
+
+        public Task OnLoadAsync(IKernel kernel)
+        {
             var kernelBase = kernel as KernelBase;
-            var directive = new Command("#!doit")
+            var directive = new Command("#!AddStrongTypes")
```

Comment:
NIT: would prefer all-lowercase/posix compatible naming conventions for magic command names and arguments